maven-plugin-tools-annotations/src/main/java/org/apache/maven/tools/plugin/extractor/annotations/JavaAnnotationsMojoDescriptorExtractor.java

```diff
@@ -288,16 +288,19 @@ protected void populateDataFromJavadoc( Map<String, MojoAnnotatedClass> mojoAnno
             MojoAnnotationContent mojoAnnotationContent = entry.getValue().getMojo();
             if ( mojoAnnotationContent != null )
             {
-                mojoAnnotationContent.setDescription( javaClass.getComment() );
+                if ( StringUtils.isEmpty( mojoAnnotationContent.getDescription() ) )
+                {
+                    mojoAnnotationContent.setDescription( javaClass.getComment() );
+                }
 
                 DocletTag since = findInClassHierarchy( javaClass, "since" );
-                if ( since != null )
+                if ( since != null && StringUtils.isEmpty( mojoAnnotationContent.getSince() ) )
                 {
                     mojoAnnotationContent.setSince( since.getValue() );
                 }
 
                 DocletTag deprecated = findInClassHierarchy( javaClass, "deprecated" );
-                if ( deprecated != null )
+                if ( deprecated != null && StringUtils.isEmpty( mojoAnnotationContent.getDeprecated() ) )
                 {
```

Comment:
New behaviour (compatible): Only set from javadoc annotation if not already set by Java annotation..